### PR TITLE
Ozone/Wayland: Update branch to version 71.0.3545.0 aka ToT

### PR DESCRIPTION
**Pull request by msisov**
The branch is based on a weekly-rebased repo located at github.com/Igalia/chromium
Currently, around 85% of patches are upstreamed. We've reworked the design of the ozone/wayland
and it's no longer based on mus aka ui service.  The work can be tracked from the ToT crbug - 
https://bugs.chromium.org/p/chromium/issues/detail?id=578890

We ported various patches to the newest design and re implemented many features. It's hard to say
the exact changes now, but this recipe is a ground to support a multiprocess mode with all the possible
wayland performance features including: frame callback, presentation feedback, partial swap, native
gpu memory buffers and etc. Though, the chromium in this recipe is built with all these features enabled,
the browser is run with --in-process-gpu flag. The reason behind this is lack of certain patches and optimizations, which I don't want to publish yet. I will update them soon.

Test-built with:
 * Yocto sumo, GCC 7 and MACHINE="raspberrypi3"

It's hard to describe why licences changed as long as the last version of the browser was
m64, which is 7 milestones behind the current m71 milestone.

Removed licence files:
- third_party/chromevox/LICENSE
- third_party/apple_sample_code/LICENSE
- third_party/deqp/LICENSE
- third_party/harfbuzz-ng/COPYING
- third_party/mesa/LICENSE
- third_party/mt19937ar/LICENSE
- third_party/shaderc/LICENSE
- third_party/vulkan/LICENSE

The explanations are taken from the last commit messages:
Changed licence files:
- third_party/SPIRV-Tools/LICENSE -  Update SPIRV-Tools build
- third_party/angle/LICENSE - Update license and readme for use downstream
- third_party/boringssl/src/LICENSE - Note licenses for support code in the top-level LICENSE file., Merge Intel copyright notice into standard
- third_party/ffmpeg/CREDITS.chromium - Updating build configs for M67 roll
- third_party/freetype/src/docs/FTL.TXT - Update or fix links to use the https protocol instead of http.
- third_party/icu/LICENSE - Update ICU to 61.1 + local patches
- third_party/protobuf/LICENSE - third_party/protobuf: Update to cf242503
- v8/LICENSE - Add antlr4 runtime library to support Torque

zlib-aarch64.patch is now added to chromium-gn, but oe-clang patch had to be rebased and is not compatible with chromium m68. Thus, there is a different patch for chromium m71

Signed-off-by: Maksim Sisov <msisov@igalia.com>

### CODE REVIEW

File: recipes-browser/chromium/chromium-gn.inc [hunk 1]
```diff
@@ -27,6 +28,7 @@ DEPENDS += " \
     glib-2.0 \
     gn-native \
     gperf-native \
+    gtk+3 \
```
[the remark this answers is not in the document]
Comment (msisov) [reply recipes-browser/chromium/chromium-gn.inc]: chromium-x11 was the single one who has had DEPENDS on it. Now, both m68 and m71 with ozone/wayland depend on the same package. Thus, I've removed it from the chromium-x11 recipe and added it here.

**Comment by msisov** [comment]
please review again

File: recipes-browser/chromium/chromium-gn.inc (hunk 1 above)
[the remark this answers is not in the document]
Comment (otavio) [reply recipes-browser/chromium/chromium-gn.inc]: Are you sure you removed from X11? Please double check.

File: recipes-browser/chromium/chromium-ozone-wayland/0001-Fix-GCC-build-remove-constexpr-for-non-resolved-comp.patch [hunk 2]
```diff
@@ -0,0 +1,71 @@
+Submitted https://crrev.com/c/1213248
```
Comment (otavio) [inline recipes-browser/chromium/chromium-ozone-wayland/0001-Fix-GCC-build-remove-constexpr-for-non-resolved-comp.patch]: Upstream-Status: Submitted [https://crrev.com/c/1213248]

Comment (otavio) [reply recipes-browser/chromium/chromium-ozone-wayland/0001-Fix-GCC-build-remove-constexpr-for-non-resolved-comp.patch]: This should be fixed in all patches. Take a look at https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines#Patch_Header_Recommendations:_Upstream-Status

File: recipes-browser/chromium/chromium-gn.inc (hunk 1 above)
[the remark this answers is not in the document]
Comment (msisov) [reply recipes-browser/chromium/chromium-gn.inc]: https://github.com/OSSystems/meta-browser/pull/170/commits/b4dab2c78d74c9a4f32725f92ba34ba734379374#diff-7d0a054cc4678ead9de545d60f95152fL20

File: recipes-browser/chromium/chromium-ozone-wayland/0001-Fix-GCC-build-remove-constexpr-for-non-resolved-comp.patch (hunk 2 above)
Comment (msisov) [reply recipes-browser/chromium/chromium-ozone-wayland/0001-Fix-GCC-build-remove-constexpr-for-non-resolved-comp.patch]: oki-doki. Sorry, I was a bit confused. Does it look better now?

File: recipes-browser/chromium/chromium-ozone-wayland/0001-Fix-GCC-build-remove-constexpr-for-non-resolved-comp.patch [hunk 3]
```diff
@@ -0,0 +1,72 @@
+Upstream-status: Submitted [https://crrev.com/c/1213248]
```
Comment (otavio) [inline recipes-browser/chromium/chromium-ozone-wayland/0001-Fix-GCC-build-remove-constexpr-for-non-resolved-comp.patch]: Upstream-**status**: Submitted [https://crrev.com/c/1213248]

Case matters. Did you read the link I sent?

Comment (msisov) [reply recipes-browser/chromium/chromium-ozone-wayland/0001-Fix-GCC-build-remove-constexpr-for-non-resolved-comp.patch]: Wait wait. Let's make it clear now.

Yes, I did.
Do you want me to have the following header:
Upstream-**S**tatus: Submitted [https://crrev.com/c/1213248]

or like this:
Upstream-**S**tatus: Submitted

blah blah

Sent to https://crrev.com/c/1213248?

Comment (otavio) [reply recipes-browser/chromium/chromium-ozone-wayland/0001-Fix-GCC-build-remove-constexpr-for-non-resolved-comp.patch]: Upstream-Status: Submitted [https://crrev.com/c/1213248]

**Comment by otavio** [comment]
Thank you for handling the comments @msisov .

@rakuco please review it and if it is fine, please merge.

File: recipes-browser/chromium/chromium-ozone-wayland/0001-Fix-GCC-build-remove-constexpr-for-non-resolved-comp.patch (hunk 3 above)
Comment (msisov) [reply recipes-browser/chromium/chromium-ozone-wayland/0001-Fix-GCC-build-remove-constexpr-for-non-resolved-comp.patch]: thanks.

**Comment by msisov** [comment]
@rakuco ping

**Comment by rakuco** [comment]
Sorry for the delay, I'll try to take a look at the PR tomorrow.

**Pull request merged**